app/_src/kubernetes-ingress-controller/guides/allowing-multiple-authentication-methods.md

```diff
@@ -9,414 +9,326 @@ for all requests without regard for whether a request has been authenticated
-content-length: 0
-server: gunicorn/19.9.0
-access-control-allow-origin: *
-access-control-allow-credentials: true
-x-kong-upstream-latency: 154
-x-kong-proxy-latency: 2
-via: kong/2.8.1
+1. Expose the service outside the Kubernetes cluster by defining Ingress rules.
+
+    ```bash
+    echo '
+    apiVersion: networking.k8s.io/v1
+    kind: Ingress
+    metadata:
+      name: demo
+      annotations:
+        konghq.com/strip-path: "true"
+    spec:
+      ingressClassName: kong
+      rules:
+      - http:
+          paths:
+          - path: /test
+            pathType: Prefix
+            backend:
+              service:
+                name: httpbin
+                port:
+                  number: 80
+    ' | kubectl apply -f -
+    ```
+    The results should look like this:
+    ```text
+    ingress.networking.k8s.io/demo created
+    ```
+1. Test these endpoints.
+
+    ```bash
+    curl -i $PROXY_IP/test/status/200
+    ```
+    The results should look like this:
+    ```text
+    HTTP/1.1 200 OK
+    Content-Type: text/html; charset=utf-8
+    Content-Length: 0
+    Connection: keep-alive
+    Server: gunicorn/19.9.0
+    Access-Control-Allow-Origin: *
+    Access-Control-Allow-Credentials: true
+    X-Kong-Upstream-Latency: 982
+    X-Kong-Proxy-Latency: 2
+    Via: kong/3.3.1
+    ```
+
+1. Create three consumers, named `consumer-1`, `consumer-2`, and `anonymous`.
+
+   Create a consumer named `consumer-1`.
+   {% include_cached /md/kic/consumer.md kong_version=page.kong_version name='consumer-1' %}
+   Create a consumer named `consumer-2`.
+   {% include_cached /md/kic/consumer.md kong_version=page.kong_version name='consumer-2' %}
+   Create a consumer named`anonymous`.
+   {% include_cached /md/kic/consumer.md kong_version=page.kong_version name='anonymous' %}
+  
+   The `anonymous` consumer does not correspond to any real user, and only serves as a fallback.
+
+## Associate plugins with the Ingress rules
+
+1. Create two plugins.
+
+    Create a plugin named `httpbin-basic-auth`.
+
+    ```bash
+    echo '
+    apiVersion: configuration.konghq.com/v1
+    kind: KongPlugin
+    metadata:
+      name: httpbin-basic-auth
+    config:
+      anonymous: anonymous
+      hide_credentials: true
+    plugin: basic-auth
+    ' | kubectl apply -f -
+    ```
+    The results should look like this: 
+    ```text
+    kongplugin.configuration.konghq.com/httpbin-basic-auth created
+    ```
+    Create a plugin named `httpbin-kay-auth`.
+    ```bash
+    echo '
+    apiVersion: configuration.konghq.com/v1
+    kind: KongPlugin
+    metadata:
+      name: httpbin-key-auth
+    config:
+      key_names:
+        - apikey
+      anonymous: anonymous
+      hide_credentials: true
+    plugin: key-auth
+    ' | kubectl apply -f -
+    ```
+    The results should look like this:
+    ```text
+    kongplugin.configuration.konghq.com/httpbin-key-auth created
+    ```
+
+1. Associate the plugins with the Ingress rule that you created.
```

Comment:
Why do we use a complete ingress here rather than `kubectl patch`?

Reply:
```
kubectl patch ingress demo -p '{"metadata":{"annotations":{"konghq.com/plugins":"httpbin-basic-auth, httpbin-key-auth"}}}'
```

Please test

Reply:
okay!